widget/radio.go

```diff
@@ -164,6 +166,58 @@ type Radio struct {
 
 	hoveredItemIndex int
 	hovered          bool
+	focused bool
+}
+
+// FocusGained is called when the Entry has been given focus.
+func (s *Radio) FocusGained() {
+	s.focused = true
+	s.Refresh()
+}
+
+// FocusLost is called when the Entry has had focus removed.
+func (s *Radio) FocusLost() {
+	s.focused = false
+	s.Refresh()
+}
+
+// Focused returns whether or not this Entry has focus.
+func (s *Radio) Focused() bool {
+	return s.focused
+}
+func (s *Radio) TypedRune(r rune) {
+
+}
+
+func (s *Radio) TypedKey(key *fyne.KeyEvent) {
+	index := -1
+	for i:=0; i<len(s.Options); i++ {
+		if s.Selected == s.Options[i] {
+			index = i;
+			break
+		}
+	}
+	if key.Name == fyne.KeyLeft || key.Name == fyne.KeyUp {
+		index--
```

Comment:
I don't think that "too far to the left" should mean "no radio selected" 
```go
if index > 1 {
    s.Selected = s.Options[index-1]
}
``` 
is cleaner

Reply:
But with the mouse, it is possible to deselect all radio buttons, so this code was added to make it possible using the keyboard alone. But generaly, I think the radio buttons should allways have one and only one button selected all the time. It should not be possible to deselect all radio buttons with the mouse.

Reply:
There has been some discussion in this area, both radio and select could do with an API that makes it explicitly possible or not to unselect